### ISSUE TITLE
Arena unfair attack point gaining and losing.

### ISSUE THREAD

**Issue by External-Amit** [PavelZinchenko/event-horizon-frontier-builds#321]
The issue is all about the attack point gaining on winning a match, you do sometimes get like 1,2,3,4,5,10,20... just like this or sometimes you get nothing, even on doing streaks of win. Why this, peoples are facing it brutally and something happened with me today, i went to 100k attack points but when i went to see it today, it was 99k. I got mad for a bit but anyways i started playing and on my first match i won but my score decreased to 98k, like wtf. WHY, it took over 4 hrs to reach 1k out of 100k and it decreased by 2k in a few minutes. WHY?

THIS IS KINDLY REQUESTED TO FIX IT AS FAST AS POSSIBLE.

And if possible please do something of my problem, it is so annoying to repeatedly gain points in that sh**ty system.

![Screenshot_2024-06-18-01-13-16-548_com.ZipasGames.Frontier.jpg](https://github.com/PavelZinchenko/event-horizon-frontier-builds/assets/158443027/04c25183-02b8-4832-bfd2-d36adcd4995c)

This was my position yesterday (17 jun)
and now you can see my position in arena.

@PavelZinchenko

**Comment by Snowgods** [comment]
It can take a little time for the  server to update, it get's there eventually.

**Comment by IsmollCheeKen** [comment]
I think I figured it out why this happens. There's a system in place where if you battle in Arena and then Load Cloud Save, you lose 1k points, and it tells you that you are cheating. This is to prevent the old exploit of loading cloud save to get back 10 Arena Battles for rapid rating farming.

The issue is, at any point you enter and battle in the Arena, you risk triggering the system if, at any point after the Arena Battle, you Load Cloud Save.

A common situation that triggers it is if you battle in Arena at least once -> go back to doing PvE Levels -> then something undesirable happens -> and you Load Cloud Save. In this case, you have unknowingly performed the exploit and has lost 1k rating.

I think the system triggers if it detects any change in Arena Battle Attempts after loading a Cloud Save. So it could also work in reverse, ex: you have 0 Arena Battles left -> Save Game to Cloud -> 30 minutes pass, gain 1 Arena Battle -> Load Cloud Save -> Arena Battle Attempts go back to 0. In this case, this will also trigger the system, causing you to lose 1k rating.

Another thing, the system also makes you lose an additional 1k rating (2k rating lost total) if you battle in Arena right after the system has been triggered.

Considering the number of Likes emote on the original post, a lot never knew about this system, nor even knew what they are doing wrong. The cases in which the system will be triggered is just way too common, you can easily trigger it, and "commit" a cheat without your knowledge and just by playing the game normally.

An anti-exploit system that ruins players' Arena ratings unknowingly. As a bonus, the real anti-cheat doesn't even work, there are cheated fleets in matchmaking. Amazing.

**Comment by External-Amit** [comment]
In my case, the second one might have happened because i never used that exploit nor the cloud loading i did. I buy arena fights by paying stars.

**Comment by External-Amit** [comment]
> It can take a little time for the  server to update, it get's there eventually.

No, it doesn't. I check my scores time to time and in sight of that, i never seen some points added suddenly to my score. But on playing matches, i lose scores. Sometimes it just shows a visual glitch but sometimes it truly minus your point by 1k.
The second last time i lost 4k points in a streak which was a glitch but the last time it actually happened and i lost 10k points in total...
The journey to leaderboard and returning on my previous score.. seems like i never reached there.

**Comment by Snowgods** [comment]
I'm actually aware of the system, I was the one who originally flagged the
exploits (Discovered while rage quitting).
There's actually two things going on. (Haven't checked the code, but this
is how I would do it anyway),
1) if the server registers a fight starting before the previous one had
finished, it flags the incomplete match as a loss. Yes this will activate
if you crash, or rage quit.
2) The server keeps track of how many combats you have remaining (easy
math), If this disagrees with the client, it sends the revised value to the
client. This does not flag as a cheat attempt (or it didn't when I tested
it). I have actually had a fight counter in the negatives when reloading
after testing some changes

**Comment by Absolutan115** [comment]
Remind me with one Scout one starbase that I can even scratch even with my ship average 3000% statistic bonus
